Lib/idlelib/squeezer.py

```diff
@@ -210,6 +214,14 @@ def reload(cls):
             type="int", default=50,
         )
 
+        # Loading the font info requires a Tk root. IDLE doesn't rely
+        # on Tkinter's "default root", so the instance will reload
+        # font info using its editor windows's Tk root.
```

Comment:
The idea here is to avoid calling `Font.measure('0')` over and over again, especially in the time-sensitive context of deciding whether to auto-squeeze. Since it can easily be computed and stored in advance, I created the new `load_font()` method, which is called by `Squeezer.__init__()` and `Squeezer.reload()`.

This could be simplified by increasing coupling: Adding logic to `PyShell` to call its squeezer's `load_font()` function after font updates, or just call its squeezer's `reload()` when appropriate.

Reply:
Also, the `EditorWindow` and `Text` classes have no `font` attribute, and I haven't found something equivalent. `Font(text, name=text.cget('font'))` is the best I could find.

Reply:
Yes, the font measure should be cached.  My concern was how to get it efficiently, though it does not matter so much in reload.  Sorry for the misdirection.  The text 'font' is an option rather than attribute, and it consists of a font tuple rather than Font instance.  I think the latter would actually be better for configuration, as well as Squeezer, but changing this would be another issue (like when mousewheel resizing is added).  I believe `Font(text, text['font']).measure('0') is what we need.  I may experiment.